cmd/armada/create/cluster/createcluster.go

```diff
@@ -0,0 +1,190 @@
+package cluster
+
+import (
+	"io/ioutil"
+	"os/user"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/gobuffalo/packr/v2"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/submariner-io/armada/pkg/cluster"
+	"github.com/submariner-io/armada/pkg/defaults"
+	kind "sigs.k8s.io/kind/pkg/cluster"
+)
+
+// CreateClusterFlagpole is a list of cli flags for create clusters command
+type CreateClusterFlagpole struct {
+	// ImageName is the node image used for cluster creation
+	ImageName string
+
+	// Wait is a time duration to wait until cluster is ready
+	Wait time.Duration
+
+	// Retain if you keep clusters running even if error occurs
+	Retain bool
+
+	// Weave if to install weave cni
+	Weave bool
+
+	// Flannel if to install flannel cni
+	Flannel bool
+
+	// Calico if to install calico cni
+	Calico bool
+
+	// Kindnet if to install kindnet default cni
+	Kindnet bool
+
+	// DeployTiller if to install tiller
+	Tiller bool
+
+	// Overlap if to create clusters with overlapping cidrs
+	Overlap bool
+
+	// Debug sets log level to debug
+	Debug bool
+
+	// NumClusters is the number of clusters to create
+	NumClusters int
+}
+
+// CreateClustersCommand returns a new cobra.Command under create command for armada
+func CreateClustersCommand(provider *kind.Provider, box *packr.Box) *cobra.Command {
+	flags := &CreateClusterFlagpole{}
+	cmd := &cobra.Command{
+		Args:  cobra.NoArgs,
+		Use:   "clusters",
+		Short: "Creates multiple kubernetes clusters",
+		Long:  "Creates multiple kubernetes clusters using Docker container 'nodes'",
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			if flags.Debug {
+				log.SetLevel(log.DebugLevel)
+				//log.SetReportCaller(true)
+			}
+
+			targetClusters, err := GetTargetClusters(provider, flags)
+			if err != nil {
+				log.Fatal(err)
+			}
+
+			var wg sync.WaitGroup
+			wg.Add(len(targetClusters))
+			for _, cl := range targetClusters {
+				go func(cl *cluster.Config) {
+					err := cluster.Create(cl, provider, box, &wg)
+					if err != nil {
```

Comment:
Both here and in L94 I think you should abort the processing of other clusters in case of failure. It's not going to be useful to have the rest of the clusters deployed, I think. Do you think differently?